tests/test_entities.py

```diff
@@ -1208,6 +1230,22 @@ def test_repository_sets_available_repositories(self):
         self.assertEqual(handler.call_count, 1)
         self.assertEqual(handler.return_value['results'], response)
 
+    def test_synchronize(self):
+        """Call
+        :meth:`nailgun.entities.Product.synchronize`
+
+        """
+        with mock.patch.object(client, 'post') as client_post:
+            with mock.patch.object(
+                entities,
+                '_handle_response',
+                return_value={'results': gen_integer()},  # not realistic
+            ) as handler:
+                response = self.product.synchronize()
+        self.assertEqual(client_post.call_count, 1)
+        self.assertEqual(handler.call_count, 1)
+        self.assertEqual(handler.return_value, response)
```

Comment:
This line seems to so useful as we are asserting that the mock library is doing its job. Maybe we can just drop this.